core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -283,7 +283,12 @@ private[spark] object CoarseGrainedExecutorBackend extends Logging {
         executorConf,
         new SecurityManager(executorConf),
         clientMode = true)
-      val driver = fetcher.setupEndpointRefByURI(arguments.driverUrl)
+
+        val driver = retry(3) {
+          fetcher.setupEndpointRefByURI(arguments.driverUrl)
+        }
```

Comment:
this changes not just for k8s, right?

Reply:
Hi @felixcheung 
It's the executor used for kubernetes but It canbe called for different resource managers.
I Labeled it as K8s because it also modifies the docker images used in k8s. Not sure what would be the best component. For instance YarnCoarseGrainedExecutorBackend extends this class.

Reply:
yes, that's what I mean. in case of shared class, the PR should tag all of them

Reply:
Hi @felixcheung 
Then this patch belongs to core (due to the retries) and Kubernetes due to the java property in the docker image. I've modified the title as such.

Reply:
Hi @felixcheung  any feedback?

Reply:
I think this can still just be simpler.
```
var driver: RpcEndpointRef = null
for (i <- 0 until 3) {
  try {
    driver = ...
  } catch {
    case e: Throwable => // continue
  }
}
require(driver != null, "...")
```

Reply:
Actually If we don't add the conditions int  the for loop, it will try to get many connections, even if they are succeeding, so I fixed the other way.

Reply:
Oops @jlpedrosa I meant to retain the condition in the loop, but otherwise I find the code right above simpler. However I like retaining the original exception. Just please pay attention to the style. `driver = ` indent is off and needs to say `if (i == nTries - 1)`

Reply:
@srowen 
Fixed the indentation.  Please double check.

Reply:
It looks the same; did you push?

Reply:
@srowen  I think it's in the right place, I'm rewriting history to clean the multiple commits. I think it is in the right place...